node/sc/bridge_account_manager.go

```diff
@@ -39,34 +43,91 @@ type accountInfo struct {
 // BridgeAccountManager manages bridge account for main/service chain.
 type BridgeAccountManager struct {
 	// TODO-Klaytn need to consider multiple bridge accounts?
-	mcAccount *accountInfo
-	scAccount *accountInfo
+	pAccount *accountInfo
+	cAccount *accountInfo
 }
 
 // NewBridgeAccountManager returns bridgeAccountManager created by main/service bridge account keys.
-func NewBridgeAccountManager(mcKey, scKey *ecdsa.PrivateKey) (*BridgeAccountManager, error) {
-	mcAcc := &accountInfo{
-		key:      mcKey,
-		address:  crypto.PubkeyToAddress(mcKey.PublicKey),
+func NewBridgeAccountManager(dataDir string) (*BridgeAccountManager, error) {
+	pWallet, pAccAddr, isLock, err := InitializeBridgeAccountKeystore(path.Join(dataDir, "parent_bridge_account"))
+	if err != nil {
+		return nil, err
+	}
+
+	if isLock {
+		logger.Warn("parent_bridge_account is locked. Please unLock the account manually for Service chain")
+	}
+
+	cWallet, cAccAddr, isLock, err := InitializeBridgeAccountKeystore(path.Join(dataDir, "child_bridge_account"))
+	if err != nil {
+		return nil, err
+	}
+
+	if isLock {
+		logger.Warn("child_bridge_account is locked. Please unLock the account manually for Service chain")
+	}
+
+	logger.Info("bridge account is loaded", "parent", pAccAddr.String(), "child", cAccAddr.String())
+
+	pAccInfo := &accountInfo{
+		wallet:   pWallet,
+		address:  pAccAddr,
 		nonce:    0,
 		chainID:  nil,
 		gasPrice: nil,
 	}
 
-	scAcc := &accountInfo{
-		key:      scKey,
-		address:  crypto.PubkeyToAddress(scKey.PublicKey),
+	cAccInfo := &accountInfo{
+		wallet:   cWallet,
+		address:  cAccAddr,
 		nonce:    0,
 		chainID:  nil,
 		gasPrice: nil,
 	}
 
 	return &BridgeAccountManager{
-		mcAccount: mcAcc,
-		scAccount: scAcc,
+		pAccount: pAccInfo,
+		cAccount: cAccInfo,
 	}, nil
 }
 
+// InitializeBridgeAccountKeystore initialize a keystore, import existing keys, and try to unlock the bridge account.
+func InitializeBridgeAccountKeystore(keystorePath string) (accounts.Wallet, common.Address, bool, error) {
+	ks := keystore.NewKeyStore(keystorePath, keystore.StandardScryptN, keystore.StandardScryptP)
+
+	// If there is no keystore file, this creates a random account and the corresponded password file.
+	if len(ks.Accounts()) == 0 {
+		pwdStr := setup.RandStringRunes(100)
```

Comment:
Just a question:
What is the meaning of the `100`?

Reply:
It is random string size.